migration/v2.20.0-ck8sx-v2.21.0-ck8s1/upgrade-cluster.md

```diff
@@ -0,0 +1,128 @@
+
+    This enables NTP service and with multiple NTP servers, specificity in sweden.
+    You can visit [www.ntppool.org](https://www.ntppool.org/zone/@) to find other ntp pools if you are in other parts of the world.
+
+    ```yaml
+    ntp_enabled: true
+    ntp_manage_config: true
+    ntp_servers:
+    - "gbg1.ntp.netnod.se iburst"
+    - "gbg2.ntp.netnod.se iburst"
+    - "lul1.ntp.netnod.se iburst"
+    - "lul2.ntp.netnod.se iburst"
+    - "mmo1.ntp.netnod.se iburst"
+    - "mmo2.ntp.netnod.se iburst"
+    - "sth1.ntp.netnod.se iburst"
+    - "sth2.ntp.netnod.se iburst"
+    - "sth3.ntp.netnod.se iburst"
+    - "sth4.ntp.netnod.se iburst"
+    - "svl1.ntp.netnod.se iburst"
+    - "svl2.ntp.netnod.se iburst"
+    ntp_timezone: "Etc/UTC"
+    ```
+
+1. Disable Pod Security Policies
+
+    1. Change the following variable in both `sc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml` and `wc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml`
+
+        ```diff
+        -podsecuritypolicy_enabled: true
+        +podsecuritypolicy_enabled: false
+        ```
+
+    1. Remove `PodSecurityPolicy` from the `kube_apiserver_enable_admission_plugins` list in both `sc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml` and `wc-config/group_vars/k8s_cluster/ck8s-k8s-cluster.yaml`
+
+        ```diff
+         kube_apiserver_enable_admission_plugins:
+        -  - "PodSecurityPolicy"
+           - "NamespaceLifecycle"
+        ```
+
+1. Download the required files on the nodes
+
+    ```bash
+    ./bin/ck8s-kubespray run-playbook sc upgrade-cluster.yml -b --tags=download
+    ./bin/ck8s-kubespray run-playbook wc upgrade-cluster.yml -b --tags=download
+    ```
+
+## Upgrade steps
+
+These steps will cause disruptions in the environment.
+
+1. Upgrade the cluster to a new kubernetes version:
+
+    ```bash
+    ./bin/ck8s-kubespray run-playbook sc upgrade-cluster.yml -b --skip-tags=download
+    ./bin/ck8s-kubespray run-playbook wc upgrade-cluster.yml -b --skip-tags=download
+    ```
+
+1. Migrate terraform state.
+
+    ```bash
+    ./migration/v2.20.0-ck8sx-v2.21.0-ck8s1/migrate-terraform-openstack-user-data.sh
+    ```
+
+    The script generated a `terraform-temp.tfstate` file with some updated values and then ran "terraform plan" with this temporary state file.
+    If terraform did not show any unexpected changes, you can run the following snippet to update the original state file for WC & SC:
+
+    ```bash
+    for CLUSTER in wc sc; do
+        pushd ${CK8S_CONFIG_PATH}/${CLUSTER}-config > /dev/null
+        mv terraform-temp.tfstate terraform.tfstate
+        popd > /dev/null
+    done
+    ```
+
+1. Lastly apply the new state to update the modules (there should be no diff in the terraform output here)
+
+    ```bash
+    ./migration/v2.20.0-ck8sx-v2.21.0-ck8s1/apply-new-state.sh
+    ```
```

Comment:
Are these only for openstack?

Reply:
Yes, I should add an extra note there :+1: